### PR TITLE
Show QR code for listener address

### PR DESCRIPTION
**Pull request by cliik**
Resolves https://github.com/mimblewimble/grin-wallet/issues/649.

To see it in action, compile and run `grin-wallet listen`. You will see your Slatepack address as usual, but now also with a scanable QR code:
```bash
$ grin-wallet listen
Password:
20220626 22:39:35.332 WARN grin_wallet_controller::controller - Starting Tor Hidden Service for API listener at address x5fnata4ge6fbj35hwojo2eynhxq5lgzr6e3q4lbd4unoxfmekbzw7qd, binding to 127.0.0.1:3415
20220626 22:39:37.720 WARN grin_wallet_controller::controller - Starting HTTP Foreign listener API server at 127.0.0.1:3415.
20220626 22:39:37.721 WARN grin_wallet_controller::controller - HTTP Foreign listener started.
20220626 22:39:37.722 WARN grin_wallet_controller::controller - Slatepack Address is: grin1ha9dqnquxy79pfma8kwfw6ycd8hsatxe37ymsutpru5dwh9vy2ps7xl420


   █▀▀▀▀▀█  █▄▄ ▀ █▄ █▄▀▀█ ▀▄█▄█ █▀▀▀▀▀█
   █ ███ █ █▀█▄██▄ █▄ █▀▀▄ █▄▀▄█ █ ███ █
   █ ▀▀▀ █ █ ▄ ▀▄▄ █ ▄▀█ █▀ ▀▀▄▄ █ ▀▀▀ █
   ▀▀▀▀▀▀▀ █ █▄█ ▀ █ ▀▄▀ ▀▄▀ ▀▄▀ ▀▀▀▀▀▀▀
   █▄█▀█▀▀▄ ▄█▀ ▀ █▄ ▀ ▀█▀ ▀█▀ ▀▄▀▀▀█▀
    ▀▄▄█▀▀▀▀▀▄▄▀   ▄ ▄ ▀▄██▀ ▄▀▀▀█ ▄▀ ▀▀
   ▀▄▀▄█▀▀ █▄▄█ ▄█ ██ ▄█▄▄█ ██▀██▀▄▀▄█▄▀
   █   ▀█▀▀█ ▀▀ ▀▄███▄  █▄▄▄ ▀█ ▄▄▀▄  ▀█
   ▄▀▀▄ █▀▀▄ ▄██ █▄▀▄▄▀▄█▄▀▄▀▄ ██▀▄▀█▀█▀
    ▀▄█  ▀▄   ▀▀ ██▀ ██▄▄▀ ▄  █ ▄██▀▀▀█▀
   ▀▀ ▀ █▀  ██▄▀▄ ▀▄▀▄█▄ █▄▀▀▀▀█▀▀▀▀ ██▀
    ▄▀▀▄ ▀█  ▄▄▄█ █▀ ▀   ▀ █ ▀██▀▀▀ █▀██
   █ ▀▄█▄▀▄▀▀ ▀▄█▀▄▄ ▀▀▀▀▀▀▀▀▄▀██▀█▀ ▀▀▀
   █  ▀█▀▀ █▄█ ▄   ▄   ▀█▄█▀▀▀▀ ▀ ▄ ▄ █▀
   ▀ ▀▀  ▀ █▄▀█ ▀  ██▀▄▀▄▀█▄▀▀▀█▀▀▀█▄▀█
   █▀▀▀▀▀█ ▄▀ █▀█▀▀██▄ ▄█▄ ▄▀  █ ▀ █  █▀
   █ ███ █ ██▀ ▄▀▀█▀ ▄ ▄█▄ ██▄▀▀██▀█▀▀▄▀
   █ ▀▀▀ █ ▀▄ ▄▀▄ █▀▀▀█▄▄▄ █ █▀▄▀ ▀    █
   ▀▀▀▀▀▀▀ ▀▀  ▀▀▀▀▀▀▀▀  ▀ ▀▀▀▀ ▀▀ ▀▀▀▀▀





```

### CODE REVIEW

**Comment by cliik** [comment]
@quentinlesceller are you the right person to ask for review/consideration of this feature?

**Comment by yeastplume** [comment]
That looks really great, thank you for this, I'll take a look at it soon!

**Comment by phyro** [comment]
I scanned the QR code in the stdout above and got what I expected :+1:

**Comment by cliik** [comment]
Force pushed to remove `qr_code` from `controller/src/lib.rs`, as I realize it wasn't needed there, but then macos builds failed in CI, so I force pushed it back :facepalm:

**Comment by cliik** [comment]
Ok, macos builds are still failing, even though nothing changed? Is there an issue with CI for macos right now?

**Comment by quentinlesceller** [comment]
Okay yes that was the issue. Once #652 is merged you can merge master into your branch to fix the CI issue.

**Comment by cliik** [comment]
@quentinlesceller I just rebased this branch onto latest master. Should be good to go (again).

**Comment by yeastplume** [comment]
All looks good, will merge over next day or so unless there are any objections

**Review by yeastplume — APPROVED** [review]

**Pull request merged**
